Sources/Amplitude/Amplitude.swift

```diff
@@ -41,10 +41,10 @@ public class Amplitude {
 
         migrateApiKeyStorages()
         migrateDefaultInstanceStorages()
-
-        if configuration.migrateLegacyData {
+        if configuration.migrateLegacyData && getStorageVersion() < .API_KEY_AND_INSTANCE_NAME {
             RemnantDataMigration(self).execute()
         }
+        migrateInstanceOnlyStorages()
```

Comment:
should we migrate instances only storages when current storage version is less than API_KEY_AND_INSTANCE_NAME?

Reply:
Yep, there is a check is in the `migrateInstanceOnlyStorages ()` method